### DISCUSSION TITLE
Functional Parameters

### DISCUSSION THREAD

**Discussion by TafeelM** [pybamm-team/PyBaMM#3466] [General]
Hello,
I want to model nmc_LGM50_Chen2020 by changing the functional parameter values, Is it possible to do so? 
There are two type of parameter values static (fixed variable which can be updated just by changing the value) and dynamic (functional parameter variable - array with multiple values)
How can I change the values of array and use the updated Functional Parameter  (such as "Negative electrode OCP [V]") and plot Voltage vs SoC 

Please help me out with this

For example -

import pybamm
import numpy as np

# Initialize parameters
parameter_values = pybamm.ParameterValues("Chen2020")

# Retrieve the OCP function or value
U_n = parameter_values["Negative electrode OCP [V]"]

# Check if U_n is callable (i.e., a function)
if callable(U_n):
    stoich = np.linspace(0, 1, 300)
    U_n_values = np.array([U_n(st) for st in stoich])
else:
    U_n_values = np.array([U_n])

# Multiply the array by 0.95
U_n_values_modified = 0.95 * U_n_values

print(U_n_values_modified)

Please let me now if its right if so how can I plot Voltage vs Soc with the upadted  "Negative electrode OCP [V]" after multiplying it with 0.95

please help me out with this.
If there are other approaches through which the functional parameters can be changed please let me know

THANK YOU

**Comment by brosaplanella** [comment]
The parameter value obtained when calling, for example `pybamm.ParameterValues("Chen2020")` are simply a dictionary (it's explained better in [this notebook](https://docs.pybamm.org/en/latest/source/examples/notebooks/getting_started/tutorial-4-setting-parameter-values.html)). Then you can just create an interpolant to the new data or fit a function to it and pass this new OCP as a parameter to the dictionary.
